pandas/core/algorithms.py

```diff
@@ -2055,13 +2056,52 @@ def sort_mixed(values):
         strs = np.sort(values[str_pos])
         return np.concatenate([nums, np.asarray(strs, dtype=object)])
 
+    def sort_tuples(values):
+        # sorts tuples with mixed values. can handle nan vs string comparisons.
+        def cmp_func(index_x, index_y):
+            x = values[index_x]
+            y = values[index_y]
+            if x == y:
+                return 0
+            len_x = len(x)
+            len_y = len(y)
+            for i in range(max(len_x, len_y)):
+                # check if the tuples have different lengths (shorter tuples
+                # first)
+                if i >= len_x:
+                    return -1
+                if i >= len_y:
+                    return +1
+                x_i_na = isna(x[i])
+                y_i_na = isna(y[i])
+                # values are the same -> resolve tie with next element
+                if (x_i_na and y_i_na) or (x[i] == y[i]):
```

Comment:
why do we need `(x_i_na and y_i_na)`? Could one of them be null?

Reply:
equality check - both of the values are the same (both na or both the same non-na value).